### ISSUE TITLE
Inline asm errors don't report file/line no information

### ISSUE THREAD

**Issue by roblabla** [rust-lang/rust#72664]
When an error within the assembly body of an inline asm happens (such as an invalid opcode), rust reports the error without pointing which inline asm instance (e.g. which file/line) caused it to happen. For instance:

```rust
#![feature(asm)]
fn main() {
    unsafe {asm!("out al, dx");}
}
```

The above "wrong" inline asm (on x86 32-bit) will report the following:

``` 
  Compiling playground v0.0.1 (/playground)
error: <inline asm>:2:2: error: invalid operand for instruction
        out al, dx
        ^
```

This makes it hard to track down where the error is caused. 

[Playground link](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=97f734fc8f8b14e8eb901939edffbf6c).

### Meta

rustc version: 1.45.0-nightly(2020-05-26 5239f5c57bb6eb9e8940)

**Comment by estebank** [comment]
This is a shame. If someone is to take this up, you will probably have to do something similar to what we do for `format!`/`println!`: https://github.com/rust-lang/rust/pull/52649.

**Comment by mark-i-m** [comment]
@estebank I believe it is fixed by #72625?

**Comment by estebank** [comment]
Yes, didn't see the link.
